src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js

```diff
@@ -0,0 +1,199 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import StartConvoAndTrackUtterances from '../../../components/webchat/startConvoAndTrackUtterances';
+
+describe('makeBotStartConvoAndTrackUtterances actions', () => {
+  // mock store
+  const middlewares = [thunk];
+  const mockStore = configureMockStore(middlewares);
+  let fakeNext;
+  let store;
+  const connectFulfilledAction = { type: 'DIRECT_LINE/CONNECT_FULFILLED' };
+  const connectSendMessage = {
+    type: 'WEB_CHAT/SEND_MESSAGE',
+    payload: { text: 'some@email.com' },
+  };
+  const directIncomingActivity = {
+    type: 'DIRECT_LINE/INCOMING_ACTIVITY',
+    payload: { activity: 'some activity' },
+  };
+
+  beforeEach(() => {
+    fakeNext = sinon.stub();
+    store = mockStore({});
+  });
+
+  it('should correctly handle "DIRECT_LINE/CONNECT_FULFILLED" with auth true', async () => {
+    // invoke startConvoAndTrackUtterances with proper curried values
+    await StartConvoAndTrackUtterances.makeBotStartConvoAndTrackUtterances(
+      'csrfToken',
+      'apiSession',
+      'apiURL',
+      'baseURL',
+      'userFirstName',
+      'userUuid',
+      true,
+    )(store)(fakeNext)(connectFulfilledAction);
+    // fake the session storage using stubs
+
+    const actions = store.getActions();
+    expect(actions.length).to.equal(2);
+    expect(fakeNext.callCount).to.equal(1);
+
+    expect(actions[0].payload.activity).to.own.include({
+      name: 'startConversation',
+    });
+    expect(actions[0]).to.own.include({
+      type: 'DIRECT_LINE/POST_ACTIVITY',
+    });
+  });
+
+  it('should correctly handle "DIRECT_LINE/CONNECT_FULFILLED" with auth false', async () => {
+    await StartConvoAndTrackUtterances.makeBotStartConvoAndTrackUtterances(
+      'csrfToken',
+      'apiSession',
+      'apiURL',
+      'baseURL',
+      'userFirstName',
+      'userUuid',
+      false,
+    )(store)(fakeNext)(connectFulfilledAction);
+
+    const actions = store.getActions();
+    expect(actions.length).to.equal(1);
+    expect(fakeNext.callCount).to.equal(1);
+
+    expect(actions[0]).to.own.include({ type: 'WEB_CHAT/SEND_EVENT' });
+    expect(actions[0].payload).to.own.include({ name: 'webchat/join' });
+  });
+
+  describe('Handling of "DIRECT_LINE/INCOMING_ACTIVITY"', () => {
+    const IS_TRACKING_UTTERANCES = 'va-bot.isTrackingUtterances';
+    const IN_AUTH_EXP = 'va-bot.inAuthExperience';
+    const RECENT_UTTERANCES = 'va-bot.recentUtterances';
+    const sandbox = sinon.createSandbox();
+    beforeEach(() => {
+      sessionStorage.clear();
```

Comment:
Do we need to clear the session before and after each test? Would just doing it after suffice?